server/kadalu_quotad/quotad.py

```diff
@@ -26,6 +29,76 @@
 # for other tools
 #
 CONFIG_FILE = "/var/lib/glusterd/kadalu.info"
+HOSTNAME = os.uname()[1]
+KADALU_PATHS = {'info', 'subvol'}
+GLUSTER_CMD = ['gluster', '--xml']
+
+
+class CmdException(Exception):
+    """
+    Exception raised when a gluster command fails in this module
+    """
+
+
+def glustercmd(cmd, xpath):
+    """
+    Issue a single gluster command with xml output and return all
+    the values from a single xpath as an array
+    """
+    proc = subprocess.Popen(GLUSTER_CMD + cmd, stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            universal_newlines=True)
+    out, err = proc.communicate()
+    if proc.returncode != 0:
+        raise CmdException((proc.returncode, out, err))
+    xml = etree.fromstring(out)
+    result = []
+    for entry in xml.findall(xpath):
+        result.append(entry.text)
+    return result
+
+
+def get_vollist():
+    """
+    Returns array of gluster volumes on this gluster server
+    """
+    return glustercmd(["volume", "list"], 'volList/volume')
+
+
+def get_bricklist(vol):
+    """
+    Returns array of bricks provided on _this_ gluster server
+    for a given volume
+    """
+    bricklist = glustercmd(["volume", "info", vol],
+                           'volInfo/volumes/volume/bricks/brick')
+    bricks = []
+    for brick in bricklist:
+        brick_match = re.match(HOSTNAME+':(.*)', brick)
```

Comment:
Good to identify local bricks using node id, multiple IPs may fail in comparison for a few cases. `/var/lib/glusterd/glusterd.info` file will have this id. (Refer parsing here: https://github.com/gluster/glustercli-python/blob/1f26c606f1b107d0e92b5ba4ba66557330a2c98a/glustercli/metrics/utils.py#L25)

Reply:
Agree, that looks smarter.
I assume we don't want kadalu to depend on glustercli?

Reply:
`glustercli` dependency is also fine if it works. I think it pulls in some extra dependencies like `paramiko` which is not useful here, but should be fine.

Reply:
Ok, great. Doing it that way saves on some code.